pkg/controller/jobframework/setup.go

```diff
@@ -92,6 +92,11 @@ func (m *integrationManager) setupControllers(ctx context.Context, mgr ctrl.Mana
 						log.Error(err, "Failed to setup controller and webhook for job framework")
 					}
 				})
+				// Webhook must be registered now.
+				// The controller-runtime silently ignores attempts to add a webhook to an already registered endpoint.
+				if err := cb.SetupWebhook(mgr, opts...); err != nil {
```

Comment:
For a while I was thinking if it is safe to register a webhook if the CRD is not installed yet, and I think it should be safe, because in this case no requests will be sent to the webhook. Thanks!